diff --git a/src/Mono.Posix/Mono.Posix.csproj b/src/Mono.Posix/Mono.Posix.csproj
index 9121d01..5978aef 100644
--- a/src/Mono.Posix/Mono.Posix.csproj
+++ b/src/Mono.Posix/Mono.Posix.csproj
@@ -3,6 +3,7 @@
     <TargetFrameworks>netstandard2.0;netstandard2.1;netcoreapp3.1;net6.0;net45</TargetFrameworks>
     <GenerateAssemblyTitleAttribute>false</GenerateAssemblyTitleAttribute>
     <IsPackable>true</IsPackable>
+    <AssemblyName>Openur.Mono.Posix</AssemblyName>
   </PropertyGroup>
 
   <Import Project="../../Mono.Unix.Common.props" />
diff --git a/src/Mono.Unix/Mono.Unix.csproj b/src/Mono.Unix/Mono.Unix.csproj
index 718c654..24fd625 100644
--- a/src/Mono.Unix/Mono.Unix.csproj
+++ b/src/Mono.Unix/Mono.Unix.csproj
@@ -4,6 +4,7 @@
     <GenerateAssemblyTitleAttribute>false</GenerateAssemblyTitleAttribute>
     <Nullable>enable</Nullable>
     <IsPackable>true</IsPackable>
+    <AssemblyName>Openur.Mono.Unix</AssemblyName>
   </PropertyGroup>
 
   <Import Project="../../Mono.Unix.Common.props" />
diff --git a/src/Mono.Unix/Mono.Unix.proj b/src/Mono.Unix/Mono.Unix.proj
index 82e43a1..3fbbc0b 100644
--- a/src/Mono.Unix/Mono.Unix.proj
+++ b/src/Mono.Unix/Mono.Unix.proj
@@ -289,5 +289,8 @@
     <Content Include="$(MSBuildThisFileDirectory)Mono.Unix.targets">
       <PackagePath>build</PackagePath>
     </Content>
+    <Content Include="$(MSBuildThisFileDirectory)Openur.Mono.Unix.targets">
+      <PackagePath>build</PackagePath>
+    </Content>
   </ItemGroup>
 </Project>
diff --git a/src/Mono.Unix/Openur.Mono.Unix.targets b/src/Mono.Unix/Openur.Mono.Unix.targets
new file mode 120000
index 0000000..52b2d2e
--- /dev/null
+++ b/src/Mono.Unix/Openur.Mono.Unix.targets
@@ -0,0 +1 @@
+Mono.Unix.targets
\ No newline at end of file
